docs/source/operator/credentials.rst

```diff
@@ -0,0 +1,56 @@
+
+   .. image:: testCredentials.jpg
+
+5. Restart the Frank!Runner but omit the Java Virtual Machine options: just ``.\start.bat``.
+6. Run the same adapter. Now it should look like shown below:
+
+   .. image:: credentialsNotAllowed.jpg
+
+   You did not allow the Frank!Framework to expand the credentials of alias ``alias1``. You see text that tells you so.
+
+The properties ``authAliases.expansion.allowed``, ``credentialFactory.class`` and ``credentialFactory.map.properties`` are system properties. You can initialize them as Java Virtual Machine properties (with ``-D`` as shown). If you run the Frank!Framework on Apache Tomcat, you can configure them in ``catalina.properties``. For other application servers, there are other ways but these are beyond the scope of this section.
+
+With property ``authAliases.expansion.allowed``, you define for which aliases you want to allow credentials expansion. It is a comma-separated list. With property ``credentialFactory.class`` you define the source from which the credentials have to be obtained. In the example the credentials were in a properties file, but there are many other possibilities you can choose. You give the Java class name of the class that should read the credentials.
+
+Depending on the value of ``credentialFactory.class``, additional properties can be needed to define the source of the credentials. If ``credentialFactory.class`` is ``nl.nn.credentialprovider.PropertyFileCredentialFactory``, you are defining that the credentials are in a properties file. In this case you should provide property ``credentialFactory.map.properties``. The value of the property is the name of the properties file where the credentials can be found.
+
+Below, all the options are listed for providing credentials
+
+Credentials from properties file
+  Use credentials factory ``nl.nn.credentialprovider.PropertyFileCredentialFactory``. Configure the properties file name in property ``credentialFactory.map.properties``.
+
+Username and password in separate text files
+  Use credentials factory ``nl.nn.credentialprovider.FileSystemCredentialFactory``. Provide the file name where the username can be found in property ``credentialFactory.filesystem.usernamefile``. Provide the file name where the password can be found in property ``credentialFactory.filesystem.passwordfile``. These file names are relative to the path you provide in property ``credentialFactory.filesystem.root``. Default values of these are ``username``, ``password`` and ``/etc/secrets``.
+
+Credentials in Ansible vault
+  Use credentials factory ``nl.nn.credentialprovider.AnsibleVaultCredentialFactory``. Provide the vault file with property ``credentialFactory.ansibleVault.vaultFile``. Provide the key file with property ``credentialFactory.keyFile``.
+
+Credentials configured in Websphere Application Server
+  User credentials factory ``nl.nn.credentialprovider.WebSphereCredentialFactory``. No additional properties needed.
```

Comment:
Het lijkt me overzichtelijker dit in een tabel te presenteren.

Reply:
Dat heb ik geprobeerd, maar dan past het niet. Ik had een tabel met vier komommen: credentials class, betekenis, extra parameter, betekenis.

Reply:
Ik ga het met twee kolommen proberen: links de trefwoorden en rechts wat meer uitleg.

Reply:
Gelukt, met dank aan StackOverflow.